Sources/TensorFlow/Context.swift

```diff
@@ -83,13 +102,51 @@ public func withContext<R>(_ context: Context, _ body: () throws -> R) rethrows
 ///   - body: A nullary closure. If the closure has a return value, that value is also used as the
 ///     return value of the `withLearningPhase(_:_:)` function.
 /// - Returns: The return value, if any, of the `body` closure.
-public func withLearningPhase<R>(_ learningPhase: LearningPhase,
-                                 _ body: () throws -> R) rethrows -> R {
+public func withLearningPhase<R>(
+    _ learningPhase: LearningPhase,
+    _ body: () throws -> R
+) rethrows -> R {
     var context = ContextManager.local.currentContext
     context.learningPhase = learningPhase
     return try withContext(context, body)
 }
 
+/// Calls the given closure within a context that has everything identical to the current context
+/// except for the given random seed.
+///
+/// - Parameters:
+///   - randomSeed: A random seed that will be set before the closure gets called and restored
+///     after the closure returns.
+///   - body: A nullary closure. If the closure has a return value, that value is also used as the
+///     return value of the `withRandomSeedForTensorFlow(_:_:)` function.
+/// - Returns: The return value, if any, of the `body` closure.
+public func withRandomSeedForTensorFlow<R>(
+    _ randomSeed: (Int32, Int32),
+    _ body: () throws -> R
+) rethrows -> R {
+    var context = ContextManager.local.currentContext
+    context.randomSeed = randomSeed
+    return try withContext(context, body)
+}
+
+/// Calls the given closure within a context that has everything identical to the current context
+/// except for the given random number generator.
+///
+/// - Parameters:
+///   - randomNumberGenerator: A random number generator that will be set before the closure gets 
+///     called and restored after the closure returns.
+///   - body: A nullary closure. If the closure has a return value, that value is also used as the
+///     return value of the `withRandomNumberGeneratorForTensorFlow(_:_:)` function.
+/// - Returns: The return value, if any, of the `body` closure.
+public func withRandomNumberGeneratorForTensorFlow<G: RandomNumberGenerator, R>(
```

Comment:
I think the `ForTensorFlow` phrase can be removed here since it is not for TensorFlow ops like the tuple seed.
```suggestion
public func withRandomNumberGenerator<G: RandomNumberGenerator, R>(
```

Reply:
The reason I kept it is because we only default to this generator in TensorFlow functions which take generators as arguments so if there is a standard library function that uses a default random number generator (which I'm not aware of but I imagine there probably are), this function will not have any effect there. So, it could cause confusion if not qualified as being specific to TensorFlow.

Reply:
My initial reasoning behind making the global variable `randomSeedForTensorFlow` was because a global variable is way more public than a type property.

In the `Context` case, since the type is already TensorFlow-specific (we can expect users to know that `Context` is from the `TensorFlow` module), I feel properties and methods don't need to mention TensorFlow. Similarly, `Context.local.randomSeed` is already without `ForTensorFlow`. WDYT?

Reply:
I see your point, although this is a global function with no reference to context in the function name or arguments. I can remove the `ForTensorFlow` and make the documentation more explicit about this. Should we also do the same for `withRandomSeedForTensorFlow`?

Reply:
Oh, I misunderstood—I thought it was a method. Keeping `ForTensorFlow` makes sense to me now.

Reply:
Sounds good! Thanks for bringing it up.